src/encoder.rs

```diff
@@ -1209,11 +1209,18 @@ pub fn encode_tx_block<T: Pixel, W: Writer>(
       residual,
```

Comment:
I think I did init the residue with zeros in case visible area is smaller than tx size.
I remember I did it when a memory is allocated for the residual, like as early as line 1184 above, i.e. instead of uninitialized, I probably used sth like initialized (if that was available? :) ).
Anyway, thanks for the fix!

Reply:
Thank you for taking a look. The primary issue was the inconsistency in stride. This was highlighted in the original TODO you left but that was removed when we reduced the zeroing. My first attempt at addressing the stride accidentally introduced an unzeroed region in the `visible_tx_w == 0` case. This is fortunately resolved now.